src/generic-container.ts

```diff
@@ -64,21 +75,33 @@ export class GenericContainer implements TestContainer {
       env: this.env,
       cmd: this.cmd,
       tmpFs: this.tmpFs,
-      boundPorts
+      boundPorts,
+      name: this.name
     });
     await this.dockerClient.start(container);
     const inspectResult = await container.inspect();
     const containerState = new ContainerState(inspectResult);
     await this.waitForContainer(container, containerState, boundPorts);
 
-    return new StartedGenericContainer(container, this.dockerClient.getHost(), boundPorts, this.dockerClient);
+    return new StartedGenericContainer(
+      container,
+      this.dockerClient.getHost(),
+      boundPorts,
+      inspectResult.name,
```

Comment:
Why not use `this.name`? That way we don't have to also add it to the `InspectResult` https://github.com/testcontainers/testcontainers-node/pull/25/files#diff-691ac3f0f06fddc7f3079b1a0f835425R12

Reply:
Containers are given 2 word names by default. And also it seems they [may contain](https://github.com/moby/moby/issues/6705#issuecomment-47298276) some prefix that *testcontainers-node* can not know about. So better the get it with inspect, rather than making the assumption the name is what we have given to the dockerode.